runtime/compiler/optimizer/IdiomRecognitionUtils.cpp

```diff
@@ -420,9 +420,7 @@ createIdiomArrayStoreBodyInLoop(TR_PCISCGraph *tgt, int32_t ctrl, int dagId, TR_
       }
    else
       {
-      i2c = new (PERSISTENT_NEW) TR_PCISCNode(tgt->trMemory(),
-                                              (opcode == TR::cstorei) ? (TR_CISCOps)TR::i2s : TR_conversion,
-                                              (opcode == TR::cstorei) ? TR::Int16 : TR::NoType,
+            i2c = new (PERSISTENT_NEW) TR_PCISCNode(tgt->trMemory(), TR_conversion, TR::NoType,
```

Comment:
I understand this is the correct contraction of the ternary - has there been verficiation done to show that this pattern matches and transforms correctly since the non cstorei case may not have matched much prior to this change...

Reply:
This code change passes jvm29 all gold tests + shows good results from all DayTrader and Liberty benchmarks (only gets a 3% drop in the `CleanedLibertyStartupDT7 - 17dev-4way-0-256-qs` test ).

Reply:
Have conduct internal testing on both functionality tests and performance benchmarks. Both are showing good results with this ternary change.